From 4d8d7ff38fd788a26cc5cdfe6e8ed60b8f4e159a Mon Sep 17 00:00:00 2001
From: 1aeo <tor@1aeo.com>
Date: Thu, 5 Mar 2026 04:51:50 +0000
Subject: [PATCH 4/5] Fix NoSectionError reference for Python 3 config parsing

Python 3's ConfigParser class does not have NoSectionError as a class
attribute. It lives on the configparser module. The except clause raises
AttributeError instead of catching the missing section error.

Triggers when ~/.exitmaprc exists but lacks a [Defaults] section.
---
 src/exitmap.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/exitmap.py b/src/exitmap.py
index 41d5d67..7418c46 100644
--- a/src/exitmap.py
+++ b/src/exitmap.py
@@ -29,6 +29,7 @@ import argparse
 import datetime
 import random
 import logging
+import configparser
 from configparser import ConfigParser
 import functools
 import pwd
@@ -122,7 +123,7 @@ def parse_cmd_args():
     if file_parsed:
         try:
             defaults = dict(config_parser.items("Defaults"))
-        except ConfigParser.NoSectionError as err:
+        except configparser.NoSectionError as err:
             log.warning("Could not parse config file \"%s\": %s",
                         config_file, err)
             defaults = {}
-- 
2.43.0

